### PR TITLE
Add Todo App API

### PR DESCRIPTION
**Pull request by MaratOM**

### CODE REVIEW

File: lesson5/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  }
+  catch(err) {
+    console.log(err)
+    res.send('Something went wrong!')
+  }
+})
+
+app.put('/todos/:id',  async (req, res) => {
+  try {
+    const message = await new Todo().updateTodo(req.params.id, req.body)
```
Comment (ubcent) [inline lesson5/index.js:36]: Looks strange. This method should be static in a current use case or as a preferable way it should have the following API:
```javascript
const todo = Todo.findById(req.params.id);
todo.updateTodo(req.body);
```

File: lesson5/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+  catch(err) {
+    console.log(err)
+    res.send('Something went wrong!')
+  }
+})
+
+app.patch('/todos/:id/complete',  async (req, res) => {
+  try {
+    const message = await new Todo().todoToggleCompleted(req.params.id)
```
Comment (ubcent) [inline lesson5/index.js:47]: The same

**Review by ubcent — APPROVED** [review]

File: lesson5/index.js (hunk 1 above)
Comment (MaratOM) [reply lesson5/index.js:36]: Agree. Will make it static.

Comment (MaratOM) [reply lesson5/index.js:36]: const todo = Todo.findById(req.params.id);
todo.updateTodo(req.body);

I don't like methods like findById to be outside of model. I believe that best way is to incapsulate all db methods in one place. So static method looks like more convenient approach in this case.

File: lesson5/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+})
+
+app.get('/todos',  async (req, res) => {
+  const todos = await Todo.find()
+  res.send(todos)
+})
+
+app.post('/todos',  async (req, res) => {
+  try {
+    const message = await new Todo().addTodo(req.body)
```
Comment (ubcent) [inline lesson5/index.js:25]:
```javascript
const todo = new Todo(req.body);
await todo.save();
```

File: lesson5/index.js (hunk 1 above)
Comment (ubcent) [reply lesson5/index.js:36]:
```javascript
Task.update({_id: req.params.id}, {$set: req.body});
```

File: lesson5/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  }
+  catch(err) {
+    console.log(err)
+    res.send('Something went wrong!')
+  }
+})
+
+app.delete('/todos/:id',  async (req, res) => {
+  try {
+    const message = await new Todo().deleteTodo(req.params.id)
```
Comment (ubcent) [inline lesson5/index.js:58]:
```javascript
Todo.delete(req.params.id);
```

File: lesson5/models/todo.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    throw new Error(error)
+  }
+}
+
+todoSchema.methods.updateTodo = async function(id, todoNewData) {
+  try {
+    const todo = await this.model('todo').findById(id)
+
+    if(todo) {
+      await todo.set(todoNewData).save()
```
Comment (ubcent) [inline lesson5/models/todo.js:39]:
```javascript
Todo.update({_id: id}, {$set: todoNewData});
```

**Review by ubcent — APPROVED** [review]

File: lesson5/models/todo.js (hunk 5 above)
Comment (MaratOM) [reply lesson5/models/todo.js:39]: Спасибо!

**Pull request merged**